sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ServiceBusReceiveLinkProcessor.java

```diff
@@ -61,6 +61,10 @@ public class ServiceBusReceiveLinkProcessor extends FluxProcessor<ServiceBusRece
     // size() on Deque is O(n) operation, so we use an integer to keep track. All reads and writes to this are gated by
     // the `queueLock`.
     private final AtomicInteger pendingMessages = new AtomicInteger();
+    // link.getCredits() is thread-unsafe, so we use an Atomic to calculate adding credits. unreceivedMessages here
+    // present for the number of messages that are not received by the ServiceBusReceiveLinkProcessor, which includes
+    // messages received by link but buffered in the reactor-operator-chain.
+    private final AtomicInteger unreceivedMessages = new AtomicInteger();
```

Comment:
whether this is count of message, or count of credit, or they are same?

Reply:
Count for the messages that haven't passed to this ServiceBusReceiveLinkProcessor, it includes the messages received but saved in the reactor buffer.

unreceivedMessages = link credits + buffered messages

Reply:
any suggestions for this variable name?